### PR TITLE
Do not make product variant's fields empty while editing simple product

### LINKED ISSUE #7735: Variant name lost when editing the product
Hi. Strange thing when editing a product with only one variant. Here are the steps to reproduce this issue.
 - Do a fresh sylius install
 - go to admin and go the product list
 - find a product with only one variant
 - list variants of that product, take note of the name of the variant
 - edit and save the product
 - go to the list of variants of the product

You'll see the variant name has changed. Further more, when you edit the variant, name fields appears blank. 

This only seem to happen when product has only one variant. 

Anyone experiencing this?

### PR DESCRIPTION
**Pull request by pamil**
| Q               | A
| --------------- | ---
| Bug fix?        | yes |
| New feature?    | no |
| BC breaks?      | no |
| Related tickets | fixes #7735 |
| License         | MIT |

TODO:
 - [x] Regression tests (null set to product variant's name after saving simple product)

### CODE REVIEW

File: src/Sylius/Bundle/AdminBundle/Resources/views/Product/Tab/_details.html.twig [hunk 1]
```diff
@@ -53,6 +56,11 @@
                 <h4 class="ui dividing header">{{ 'sylius.ui.taxes'|trans }}</h4>
                 {{ form_row(form.variant.taxCategory) }}
             </div>
+            <div class="ui segment">
+                <h4 class="ui dividing header">{{ 'sylius.ui.availability'|trans }}</h4>
+                {{ form_row(form.variant.availableOn) }}
+                {{ form_row(form.variant.availableUntil) }}
```
Comment (plozmun) [inline src/Sylius/Bundle/AdminBundle/Resources/views/Product/Tab/_details.html.twig]: This should only be shown on simple products

**Comment by plozmun** [comment]
To prevent setting name to null on save, yo can add 
`$form->get('variant')->remove('name');` 
at this line 
https://github.com/Sylius/Sylius/blob/master/src/Sylius/Bundle/ProductBundle/Form/EventSubscriber/SimpleProductSubscriber.php#L51-L56

File: src/Sylius/Bundle/AdminBundle/Resources/views/Product/Tab/_details.html.twig (hunk 1 above)
Comment (pamil) [inline src/Sylius/Bundle/AdminBundle/Resources/views/Product/Tab/_details.html.twig]: It is, see line 41 :)

Comment (pamil) [inline src/Sylius/Bundle/AdminBundle/Resources/views/Product/Tab/_details.html.twig]: It is, see line 41 :)

Comment (plozmun) [inline src/Sylius/Bundle/AdminBundle/Resources/views/Product/Tab/_details.html.twig]: Ok, i see, sorry

**Comment by pjedrzejewski** [comment]
Thanks Kamil! 👍 👍 👍

**Pull request merged**